source/Ship.cpp

```diff
@@ -2263,36 +2263,48 @@ int Ship::Scan()
 	if(!cargoDistance && !outfitDistance)
 		return 0;
 	
-	// Scanning speed also uses a square root, so you need four scanners to get
-	// twice the speed out of them.
-	double cargoSpeed = sqrt(attributes.Get("cargo scan speed"));
+	// Scanning speed does not use a square root, so that increasing both power and speed 
+	// proportionally will rise both range and speed. (The longer the range, the slower the scanner.)
+	// Multiply by 100 so that we can simply divide speed by power.
+	double cargoSpeed = 100. * attributes.Get("cargo scan speed") / cargoDistance;
 	if(!cargoSpeed)
-		cargoSpeed = 1.;
-	double outfitSpeed = sqrt(attributes.Get("outfit scan speed"));
+		cargoSpeed = 100. / cargoDistance;
+	
+	double outfitSpeed = 100. * attributes.Get("outfit scan speed") / outfitDistance;
 	if(!outfitSpeed)
-		outfitSpeed = 1.;
+		outfitSpeed = 100. / outfitDistance;
 	
 	// Check how close this ship is to the target it is trying to scan.
 	double distance = (target->position - position).Length();
 	
+	// Check the target's outfit and cargo space, a larger ship takes longer to scan.
+	// Normalized around 200 tons of cargo/outfit space.
+	double outfits = target->baseAttributes.Get("outfit space") / 200.;
+	double cargo = target->attributes.Get("cargo space") / 200.;
+	
 	// Check if either scanner has finished scanning.
 	bool startedScanning = false;
 	bool activeScanning = false;
 	int result = 0;
-	auto doScan = [&](double &elapsed, const double speed, const double scannerRange, const int event) -> void
+	auto doScan = [&](double &elapsed, const double speed, const double scannerRange, double depth, const int event) -> void
 	{
 		if(elapsed < SCAN_TIME && distance < scannerRange)
 		{
 			startedScanning |= !elapsed;
 			activeScanning = true;
+			elapsed +=
+				min(1., 2. * (scannerRange - distance) / scannerRange)
+				* (5. / (5. + (Velocity() - target->Velocity()).Length()))
+				* (depth / (depth + target->attributes.Get("scan interference")))
```

Comment:
Understandable. However, the way interference works now doesn't feel very impactful... This is a very tangible, and potentially very easy to take advantage of, implementation, but you're right in that it doesn't make much sense physically (and indeed it would be pretty obvious something was up if you knew how long it usually takes to scan something). I'll have to think up a better implementation, if this doesn't fly.

Reply:
If scan completion time varies based on mass of target, speed of scanner, and interference, it wouldn't bemuch of a red flag. Especially if there is a variety of scanners available.